trace/doc.go

```diff
@@ -0,0 +1,164 @@
+// Package trace provies an API for initiating and reporting SSF
+// traces and attaching spans to them. Veneur's tracing API also
+// provides an experimental opentracing compatibility layer.
+//
+// The Veneur tracing API is completely independent of the opentracing
+// compatibility layer, with the exception of the function
+// StartSpanFromContext.
+//
+// Setup and Structure of the API
+//
+// To use the API, user code must set its service name via the global
+// variable Service, which is used by trace visualization tools to
+// tell apart the spans reported by different services. Code using the
+// trace API must set its Service, and set it only once, in main():
+//
+//    trace.Service = "my-awesome-service"
+//
+// The trace API is built on three abstractions: the
+// Client, the ClientBackend and the Span.
+//
+// Context
```

Comment:
Slight note: it's a little jarring to say "the API is built on three abstractions" and then immediately jump to a new section that is titled with a different thing (not one of those three). I don't know how to resolve this, but it jumped out at me, reading with fresh eyes.

Reply:
Lol, it's true. If we could have subsections in godoc, that would be less of an issue, I believe. I'll try and rephrase this and the following sections.